File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char, int> mpp;
        int start = -1;

        for (int i = 0; i < s.size(); i++) {
            mpp[s[i]]++;
        }

        for (int i = 0; i < s.size(); i++) {
            if (mpp[s[i]] == 1) {
                start = i;
                break;
            }
        }

        return start;
    }
};

File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        mpp = {}
        for x in s:
           mpp[x]=mpp.get(x,0)+1
        for i in range(len(s)):
            if mpp[s[i]]==1:
                return i
        return -1        
